static/app/views/discover/resultsSearchQueryBuilder.tsx

```diff
@@ -0,0 +1,297 @@
+      if (isDeviceClass(tag.key)) {
+        return Promise.resolve(DEVICE_CLASS_TAG_VALUES);
+      }
+      const fetchPromise =
+        dataset === DiscoverDatasets.SPANS_INDEXED
+          ? fetchSpanFieldValues({
+              api,
+              endpointParams: normalizeDateTimeParams(selection.datetime),
+              orgSlug: organization.slug,
+              fieldKey: tag.key,
+              search: query,
+              projectIds: projectIdStrings,
+            })
+          : fetchTagValues({
+              api,
+              endpointParams: normalizeDateTimeParams(selection.datetime),
+              orgSlug: organization.slug,
+              tagKey: tag.key,
+              search: query,
+              projectIds: projectIdStrings,
+              // allows searching for tags on transactions as well
+              includeTransactions: includeTransactions,
+              // allows searching for tags on sessions as well
+              includeSessions: includeSessionTagsValues,
+              dataset: dataset ? DiscoverDatasetsToDatasetMap[dataset] : undefined,
+            });
+
+      try {
+        const results = await fetchPromise;
+        return results.filter(({name}) => defined(name)).map(({name}) => name);
+      } catch (error) {
+        throw new Error('Unable to fetch event field values');
+      }
+    },
+    [
+      api,
+      organization,
+      selection.datetime,
+      projectIds,
+      includeTransactions,
+      includeSessionTagsValues,
+      dataset,
+    ]
+  );
+
+  const getTagList: TagCollection = useMemo(() => {
+    const measurementsWithKind = getMeasurementTags(measurements, customMeasurements);
+    const orgHasPerformanceView = organization.features.includes('performance-view');
+
+    const combinedTags: TagCollection =
+      dataset === DiscoverDatasets.ERRORS
+        ? Object.assign({}, functionTags, STATIC_FIELD_TAGS_WITHOUT_TRANSACTION_FIELDS)
+        : dataset === DiscoverDatasets.TRANSACTIONS ||
+            dataset === DiscoverDatasets.METRICS_ENHANCED
+          ? Object.assign(
+              {},
+              measurementsWithKind,
+              functionTags,
+              STATIC_SPAN_TAGS,
+              STATIC_FIELD_TAGS_WITHOUT_ERROR_FIELDS
+            )
+          : orgHasPerformanceView
+            ? Object.assign(
+                {},
+                measurementsWithKind,
+                functionTags,
+                STATIC_SPAN_TAGS,
+                STATIC_FIELD_TAGS
+              )
+            : Object.assign({}, STATIC_FIELD_TAGS_WITHOUT_TRACING);
+
+    Object.assign(combinedTags, filteredTags, STATIC_SEMVER_TAGS);
+
+    combinedTags.has = getHasTag(combinedTags);
+
+    return combinedTags;
+  }, [
+    measurements,
+    dataset,
+    customMeasurements,
+    functionTags,
+    filteredTags,
+    organization.features,
+  ]);
+
+  const filterKeySections = useMemo(() => {
```

Comment:
This can be sorted out later, but I wonder if we should have a "Functions" section for things like `count()` and such

Reply:
Good idea, will suggest this to my teammates.